src/main/scala/net/cakesolutions/CakeStandardsPlugin.scala

```diff
@@ -33,55 +33,106 @@ object CakeStandardsPlugin extends AutoPlugin {
   override val projectSettings: Seq[Setting[_]] =
     ScalafmtCorePlugin.projectSettings ++
       Seq(Compile, Test).flatMap(inConfig(_)(scalafmtSettings)) ++
+      wartRemoverSettings ++
+      customJavacOptions ++
       Seq(
-        scalacOptions ++=
-          Seq(
-            "-encoding",
-            "UTF-8",
-            "-feature",
-            "-deprecation",
-            "-unchecked",
-            "-language:postfixOps",
-            "-language:implicitConversions",
-            "-Xlint",
-            "-Yno-adapted-args",
-            "-Ywarn-dead-code",
-            "-Ywarn-value-discard",
-            "-Ywarn-numeric-widen",
-            "-Xfuture"
-          ) ++ {
-            CrossVersion.partialVersion(scalaVersion.value) match {
-              case Some((2, 12)) =>
-                Seq("-Ywarn-unused-import", "-Ypartial-unification")
-              case Some((2, 11)) =>
-                Seq(
-                  "-Yinline-warnings",
-                  "-Ywarn-unused-import",
-                  "-Ypartial-unification"
-                )
-              case Some((2, 10)) =>
-                Seq("-Yinline-warnings")
-              case _ =>
-                Nil
-            }
-          } ++
-            // fatal warnings can get in the way during the DEV cycle
-            sys.env
-              .get("CI")
-              .fold(Seq.empty[String])(_ => Seq("-Xfatal-warnings")),
-        javacOptions ++=
-          Seq(
-            "-Xlint:all",
-            "-Xlint:-options",
-            "-Xlint:-path",
-            "-Xlint:-processing"
-          ) ++
-            sys.env.get("CI").fold(Seq.empty[String])(_ => Seq("-Werror")),
-        // some of those flags are not supported in doc
-        javacOptions in doc ~= (_.filterNot(_.startsWith("-Xlint"))),
-        // http://www.wartremover.org
-        wartremoverExcluded in Compile ++= (managedSources in Compile).value,
-        wartremoverWarnings in (Compile, compile) :=
-          Warts.unsafe ++ Seq(Wart.FinalCaseClass, Wart.ExplicitImplicitTypes)
+        scalacOptions ++= commonScalacOptions ++ {
+          CrossVersion.partialVersion(scalaVersion.value) match {
+            case Some((2, 12)) =>
+              scalacOptionsFor212
+            case Some((2, 11)) =>
+              scalacOptionsFor211
+            case _ =>
+              Nil
+          }
+        } ++
+          // fatal warnings can get in the way during the DEV cycle
+          sys.env
+            .get("CI")
+            .fold(Seq.empty[String])(_ => Seq("-Xfatal-warnings"))
```

Comment:
Lets invert this logic here - i.e. lets have fatal warnings on by default and trigger switching them off based on an environment variable or (better) a system property?

Reply:
Not sure there is much point in a switch on compiler strictness.  Id prefer to see fatal-warnings optional at the project level.  i.e. not a global default.

Reply:
Let's keep it unchanged in this PR. If we decide to change it still can do in another PR

Reply:
https://github.com/cakesolutions/sbt-cake/issues/54 is tracking this issue, so there's no need for a change here.